help_to_heat/frontdoor/views.py

```diff
@@ -805,7 +812,102 @@ def get_context(self, request, session_id, *args, **kwargs):
+            "I do not know",
+        ]
+        is_loft_insulation_under_threshold = session_data["loft_insulation"] in [
+            "I have up to 100mm of loft insulation",
+            "I do not know",
+        ]
+
+        text_flags = {
+            "park_home_text": is_in_park_home and not is_social_housing,
+        }
+        text_flags.update(
+            {
+                "cavity_wall_text": text_flags.get("park_home_text") is False
+                and is_cavity_walls
+                and not is_wall_insulation_present,
+                "solid_wall_text": text_flags.get("park_home_text") is False
+                and is_solid_walls
+                and not is_wall_insulation_present,
+                "room_in_roof_text": text_flags.get("park_home_text") is False and not is_loft_present,
+                "loft_insulation_text": text_flags.get("park_home_text") is False
+                and is_loft_present
+                and is_there_access_to_loft,
+                "contribution_text": text_flags.get("park_home_text") is False
+                and (not is_social_housing)
+                and is_not_on_benefits
+                and is_income_above_threshold,
+            }
+        )
+        text_flags.update(
+            {
+                "loft_insulation_low_contribution_text": text_flags["loft_insulation_text"] is True
+                and is_loft_insulation_under_threshold,
+                "loft_insulation_medium_contribution_text": text_flags["loft_insulation_text"] is True
+                and is_loft_insulation_over_threshold,
+            }
+        )
+
+        return {"eligible_schemes": eligible_schemes, "supplier_name": supplier_name, "text_flags": text_flags}
+
+    def validate(self, request, session_id, page_name, data, is_change_page):
+        session_data = interface.api.session.get_session(session_id)
+        fields = page_compulsory_field_map.get(page_name, ())
+        missing_fields = tuple(field for field in fields if not data.get(field))
+        errors = {field: missing_item_errors[field] for field in missing_fields}
+        is_gbis_eligible = "GBIS" in session_data["schemes"]
+        is_not_park_home = session_data.get("park_home", "No") == "No"
+        is_below_income_threshold = session_data.get("household_income") == "Less than £31,000 a year"
+        is_on_benefits = session_data.get("benefits") == "Yes"
+        is_social_housing = session_data.get("own_property") == "No, I am a social housing tenant"
+        if (
+            is_gbis_eligible
+            and is_not_park_home
+            and (is_social_housing or (is_on_benefits or is_below_income_threshold))
+        ):
+            if not data.get("ventilation_acknowledgement"):
+                errors = {
+                    **errors,
+                    "ventilation_acknowledgement": missing_item_errors["ventilation_acknowledgement"],
+                }
+        elif is_gbis_eligible:
```

Comment:
this logic is confusing me; if the user is not gbis eligible then they shouldn't be able to submit this page? should we do something else in this case?